projects/hathor-wallet-mobile/0002-nano-contract-integration.md

```diff
@@ -0,0 +1,476 @@
+  
+- **Functionality**:
+	- The *Nano Contract List* display option should reveal all registered nano contracts in the wallet when selected.
+	- The "Register Token" button navigates the user to the Token Registration screen.
+	- The "Register Nano Contract" button navigates the user to the **Nano Contract Registration** screen for adding a new Nano Contract.
+  
+- **Visibility Conditions**:
+	- The *Nano Contract List* display option should only be visible and active for users who have opted into the nano-contract feature.
+	- The "Register Nano Contract" button should be visible only when the *Nano Contract List* is being displayed.
+  
+- **Error Management**:
+  - Ensure that error messages are clear and provide guidance on how to resolve the issue or contact support.
+
+### Component: Nano Contract List
+[component-nano-contract-list]: #component-nano-contract-list
+
+This component displays a list of registered *Nano Contracts* on the *Main* screen, allowing users to select and interact with each contract. Each list item is designed to be actionable and includes two key pieces of information: the *Nano Contract ID* and the *Blueprint Name*.
+
+Tapping on a list item navigates the user to the **Nano Contract Transactions** screen.
+
+##### Design Bill
+- **Placement**:
+	- The *Nano Contract List* should be prominently displayed within the upper section of the *Main* screen for easy access.
+
+- **Functionality**:
+	- Implement an infinite scrolling view to display a potentially large number of nano contracts.
+	- Make each list item actionable, leading to the **Nano Contract Transactions** screen upon selection.
+	- Each item list item must contain *Nano Contract ID* and *Blueprint Name* information.
+
+- **Visibility Conditions**:
+	- The list should only display *Nano Contracts* that the user has registered.
+	- The *Nano Contract ID* can be shortened to fit in the screen
+  
+>[!NOTE]
+>To accommodate the *Nano Contract ID* on mobile screens, it is presented in a shortened format, displaying the first 7 and last 7 characters, e.g., `000001e...3c9641d`. The full ID can be viewed on tap or hover.
+
+### Screen: Nano Contract Transactions
+[screen-nano-contract-transactions]: #screen-nano-contract-transactions
+
+This screen is dedicated to displaying a list of *Nano Contract Transaction* items associated with a selected *Nano Contract*. After a user selects a *Nano Contract* from the *Nano Contract List* component, they are directed to this screen, where all transactions involving the chosen contract are listed.
+
+The header section of this screen provides essential state information about the selected *Nano Contract*, including the full *Nano Contract ID*, *Blueprint Name*, and a breakdown of each *Token Balance*. This header is designed to be both expandable and collapsible: in its expanded form, it reveals all state details, while in its collapsed form, it displays only the *Nano Contract ID* for a more streamlined view.
```

Comment:
Is the blueprint name unique in the network?

Should we also display an URL to the explorer?

Reply:
> Is the blueprint name unique in the network?

No, the name isn't unique.

> Should we also display an URL to the explorer?

We can have this functionality at this screen, sure.